### ISSUE TITLE
intellij: "search everywhere" does not understand static symbols

### ISSUE THREAD

**Issue by yjbanov** [dart-lang/sdk#30580]
The "search everywhere" feature (invoked by double-Shift tap) can search symbols across the project. However, it currently does not understand Dart's static symbols. For example, Flutter has a static `of` dependency injection pattern. Dozens of classes implement it. Today, I tried going to the definition of `Theme.of` but found no way to express it in my search query. I can navigate to the `Theme` class first then search within the file, but the class declaration and the declaration of `of` are quite far apart, so I waste several seconds every time I'm looking for a static symbol. Dartdoc supports `CLASS.SYMBOL` syntax (try searching for `Theme.of` and `Scaffold.of` in docs.flutter.io, works great). I think Intellij should support that too.

**Comment by jwren** [comment]
@devoncarew 
@alexander-doroshko 

I don't think that the IJ indexer supports this.  With Java which has a complete IJ indexer, SomeClass.someMethod doesn't have any hits either.  Given all the functionality in the search dialogs and the ability to filter by directory in the file dialog, I am a bit surprised that this doesn't work in IJ.  Any thoughts @alexander-doroshko?  Will this happen to be fixed in 2017.3?

**Comment by alexander-doroshko** [comment]
@yjbanov try 'Navigate Symbol' (Cmd+Alt+O), it should understand `Theme.of`

P.S. Dart plugin issue tracker is https://youtrack.jetbrains.com/issues/WEB (Subsystem: Dart), if you think that you've found plugin-specific issue / feature request feel free to file it directly there.

**Comment by yjbanov** [comment]
Thanks, Cmd+Alt+O works. I'll try to use if I manage to memorize the shortcut. Still, I'd expect a feature called "search _everywhere_" to find it too 😃

**Comment by helin24** [comment]
It seems double Shift -> searching "Theme.of" still does not show the same results, but I suspect we won't be working on this unless it becomes a high priority in the Dart plugin (can refile there if it is a problem - https://github.com/flutter/dart-intellij-third-party
